### PR TITLE
Feat/cli minor improve

### PR DESCRIPTION
**Pull request by deveaud-m**
This patch polishes the current CLI implementation before it will be released:
* update python dependencies and add some development tools
* test the CLI also with the latest stable Fossology release
* fix a bug when using older versions of the API where username was required to login
* improve aesthetic of the CLI help texts and the documentation `README.rst`

### CODE REVIEW

**Comment by deveaud-m** [comment]
@thsetz once this branch is merges to `master` CLI tests should also run for your other feature branch:https://github.com/fossology/fossology-python/tree/feat/cli-issues82and83.

Could you look at the changes here and give me feedback?

File: README.rst [hunk 1]
```diff
@@ -80,86 +80,91 @@ Using the API
...
 
-Runs the same command with verbosity level  2.
-It leaves a log-file in directory .foss_cli_results named .foss_cli.log.
+   .. code:: bash
 
-This way ALL cmds could be logged.
+      $ foss_cli -vv [COMMAND]
 
-To  create  a group:
+   This runs the given command with verbosity level 2 (all warning will be logged).
```
Comment (thsetz) [inline README.rst:120]: .....   all debug statements will be logged

**Review by thsetz — APPROVED** [review]
Looks good

**Comment by deveaud-m** [comment]
Suggestion integrated, pipeline successful, merging! Thanks @thsetz for the review.

**Pull request merged**